lib/externals/scms/git_project.rb

```diff
@@ -3,7 +3,16 @@
 module Externals
   class GitProject < Project
     def default_branch
-      'master'
+      branch = begin
+                 `git config --global init.defaultBranch`
+               rescue
+                 ""
+               end 
+      if branch == "" then
+           raise "Git default branch not configured.\n" \
+                 "Use `git config --global init.defaultBranch <default branch name>` to set one."
```

Comment:
I think we should use `master` as a default here so people without a default will not get an unexpected exception upon upgrading.

Reply:
I considered that. The reason I chose to go this route is that the community is moving away from using `master` as the default branch name. I think that this message is a reasonably neutral position on the issue, that does not explicitly encourage the use of `master` while also not enforcing `main` or `trunk`.

Reply:
Got it, understood. I think the challenge there though is I would prefer to not break existing code for people through an upgrade. I know a handful of people used to use this as part of some kind of chef deployment. Having to scramble to fix a deployment would be bad. Technically people should test their deployment before upgrading dependencies but it could bite somebody. So I think preserving backwards-compatibility is important. But see my comment below... I can't actually figure out what code path enters this method in the first place and wonder if it's unreachable.